addons/Line_Number/src/presenter.js

```diff
@@ -2380,33 +2388,141 @@ function AddonLine_Number_create() {
+                $(this).addClass('show-answers');
+            });
+        } else {
+            if (!presenter.currentRanges) {
+                presenter.saveUserRanges();
+                presenter.removeRanges();
+            }
+
+            const range = [presenter.configuration.otherRanges[eventData.item]];
+
+            presenter.drawRanges(presenter.configuration.shouldDrawRanges, true);
+            presenter.drawRanges(range, true);
+
+            $.each(presenter.configuration.drawnRangesData.ranges, function() {
+                getSelectedRange(this).addClass('show-answers');
+            });
+        }
+
+        presenter.$view.find('.outer').addClass("disable");
+        presenter.isProcessing = false;
+    }
+
+    presenter.gradualHideAnswers = function () {
+        if(!presenter.configuration.isActivity || !presenter.isGradualShowAnswersActive) {
+            return;
+        }
+
+        presenter.$view.find('.show-answers').removeClass('show-answers');
+        presenter.isGradualShowAnswersActive = false;
+
+        if (presenter.configuration.dontShowRanges) {
+            drawPoints(presenter.checkedPoints);
+            presenter.isFirstRun = true;
+        } else {
+            presenter.removeRanges();
+            presenter.redrawRanges(presenter.currentRanges.ranges);
+            presenter.displayUserRanges();
+        }
+
+        presenter.$view.find('.outer').removeClass("disable");
+        presenter.currentRanges = null;
+    }
+
+    presenter.saveUserRanges = function () {
+        presenter.currentRanges = jQuery.extend(true, {}, presenter.configuration.drawnRangesData);
+
+        presenter.leftShowAnswers = presenter.parentLeft;
+        if (presenter.leftShowAnswers) {
+            presenter.$view.find('.rangeImage').each(function () {
+                if (parseInt($(this).parent()[0].style.left, 10).toFixed(1) == parseInt(presenter.leftShowAnswers, 10).toFixed(1)) {
+                    presenter.rangeShowAnswers = $(this);
+                }
+            });
+        }
+    }
+
+    presenter.removeRanges = function () {
+        presenter.$view.find('.rangeImage').remove();
+        const rangesToRemove = [].concat(presenter.configuration.drawnRangesData.ranges);
+        $.each(rangesToRemove, function() {
+            presenter.removeRange(this, true);
+        });
+    }
+
+    presenter.displayUserRanges = function () {
+        if(presenter.leftShowAnswers){
+            presenter.$view.find('.stepLine').each(function () {
+                if(parseInt($(this)[0].style.left, 10).toFixed(1) == parseInt(presenter.leftShowAnswers, 10).toFixed(1)){
+                    $(this).append(presenter.rangeShowAnswers);
+                }
+            });
+        }
+    }
+
+    function addListenerWithTimeout() {
+        setTimeout(() => {
+            presenter.addListenerOnHover();
+        }, 200);
+    }
+    
+    presenter.addListenerOnHover = function () {

```

Comment:
Wydaje mi się, że jeżeli w GSA jest to dodawane to w GHA powinno być usuwane.  Proszę o komentarz